File: Apps/Codaxy.Dextop.Showcase/Demos/Api/FirstApiDemo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Codaxy.Dextop.Api;

namespace Codaxy.Dextop.Showcase.Demos.Remoting
{
    [Demo("FirstApiDemo",
        Title = "First API Demo",
        Description = "Show how to use stateless Dextop API",
        ClientLauncher = true,
        Path = "~/Demos/Api"
    )]
    [LevelBeginner]
    [TopicDextopRemoting]
    [CategoryDemo]
    public class FirstApiDemo : IDextopApiController
    {
        public string HelloWorld()
        {
            return "Hello World!";
        }
    }
}

File: Apps/Codaxy.Dextop.Showcase/Demos/Api/FirstApiDemo.js
Ext.ns('Showcase');

Ext.define('Showcase.demos.Launcher.FirstApiDemo', {
    extend: 'Showcase.demos.Launcher',
    launch: function () {
        var remoteId = Ext.id();
        var w = Ext.create('Showcase.demos.FirstApiDemoWindow', {});
        w.show();        
    }
});

Ext.define('Showcase.demos.FirstApiDemoWindow', {
	extend: 'Ext.window.Window',	
	width: 300,
	height: 200,
	
	title: 'Dextop API demo',	
	
	initComponent: function () {

	    this.api = Ext.create('Showcase.demos.FirstApiDemo', {

	    });

	    Ext.apply(this, {
            bodyPadding: 10,
	        items: [{
	            xtype: 'button',
	            text: 'Invoke',
	            handler: function () {
	                this.api.Hello({
	                    success: function (result) {
	                        Dextop.infoAlert(result);
	                    }
	                });
	            },
	            scope: this
	        }, {
	            xtype: 'button',
	            text: 'Info',
	            handler: function () {
	                this.api.Info({});
	            },
	            scope: this
	        }, {
	            xtype: 'button',
	            text: 'Error',
	            handler: function () {
	                this.api.Error({});
	            },
	            scope: this
	        }]
	    });

	    this.callParent(arguments);
	}
});
